benchmark_test.go

```diff
@@ -0,0 +1,65 @@
+package amqp_test
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/Azure/go-amqp"
+)
+
+func BenchmarkSimple(b *testing.B) {
+	if localBrokerAddr == "" {
+		b.Skip()
+	}
+	client, err := amqp.Dial(localBrokerAddr)
+	if err != nil {
+		b.Fatal(err)
+	}
+	defer client.Close()
+
+	session, err := client.NewSession()
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	// add a random suffix to the link name so the test broker always creates a new node
+	targetName := fmt.Sprintf("BencmarkSimple %d", rand.Uint64())
+
+	sender, err := session.NewSender(
+		amqp.LinkTargetAddress(targetName),
+	)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	receiver, err := session.NewReceiver(
+		amqp.LinkSourceAddress(targetName),
+	)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	msg := amqp.NewMessage([]byte("test message"))
+	for i := 0; i < b.N; i++ {
+		// simple send and receive message, no concurrency
```

Comment:
Does the test AMQP broker we have support a bit more than this? Ie, can we query after we've accepted the message and see the state?

Mostly just wondering how we verify correctness for our library (along with the speed).

Reply:
Not sure.  Is there an example API you're looking for?